core/shared/src/main/scala/fs2/internal/CompileScope.scala

```diff
@@ -178,6 +226,11 @@ private[internal] final class CompileScope[F[_]] private (val id: Token, private
     go(self, Catenable.empty)
   }
 
+  /** yields to true, if this scope has ancestor with given scope Id **/
+  def hasAncestor(scopeId: Token): F[Boolean] = {
+    F.map(ancestors) { c => Catenable.instance.exists(c)(_.id == scopeId) }
```

Comment:
Note that if you import `cats.implicits._`, you can write this line as `ancestors.map(_.exists(_.id == scopeId))`

Reply:
There's quite a lot of `F.map` and `F.flatMap`, but imho it hurts readability a bit due to all the parenthesis, so unless there's a specific reason not to, definitely 👍 to use cats syntax

Reply:
Are we sure there is no extra allocation coming from the cats syntax implicit?

Reply:
No, there's an allocation involved. I'm fine with the policy of `fs2.internal` not using cats syntax.